src/main/java/mate.academy/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.model.Product;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+
+public class ProductServiceImpl implements ProductService {
+    private ProductParser productParser;
+    private FileReaderService fileReaderService;
+
+    @Override
+    public List<Product> getAllFromFile(String filePath) {
+        return fileReaderService.readFile(filePath)
+                .stream()
+                .map(productParser::parse)
+                .collect(Collectors.toList());
```

Comment:
Im sorry, but is this reference or is it an actual students task?

Reply:
This is a part of the task. There is no need for students to implement this part of the functionality. Students should be focused on the Injector implementation.